plugin-server/src/worker/ingestion/timestamps.ts

```diff
@@ -4,24 +4,32 @@ import { DateTime, Duration } from 'luxon'
 
 import { status } from '../../utils/status'
 
-type InvalidTimestampCallback = () => void
+type InvalidTimestampCallback = (field: string, value: string, reason: string) => void
 
 export function parseEventTimestamp(data: PluginEvent, callback?: InvalidTimestampCallback): DateTime {
-    const now = DateTime.fromISO(data['now']).toUTC()
-    const sentAt = data['sent_at'] ? DateTime.fromISO(data['sent_at']).toUTC() : null
+    const now = DateTime.fromISO(data['now']).toUTC() // now is set by the capture endpoint and assumed valid
+
+    let sentAt: DateTime | null = null
+    if (data['sent_at']) {
+        sentAt = DateTime.fromISO(data['sent_at']).toUTC()
+        if (!sentAt.isValid) {
+            callback?.('sent_at', data['sent_at'], sentAt.invalidExplanation ?? '')
```

Comment:
@tiina303 what would be a better alternative to `?? ''` to handle null and undefined?